crates/re_ui/src/list_item.rs

```diff
@@ -16,6 +17,18 @@ pub struct ShowCollapsingResponse<R> {
     pub body_response: Option<R>,
 }
 
+/// Specification of how the width of the [`ListItem`] must be allocated.
+#[derive(Default, Clone, Copy, Debug)]
+pub enum WidthAllocationMode {
+    /// Allocate the full available width. Useful for fixed-width container.
+    #[default]
+    Available,
+
+    /// Allocate the width needed for the text and icon(s) (if any). If buttons are used, the label
+    /// will get truncated to display them.
+    Fit,
```

Comment:
When reading the use of this, I interpreted it to mean "Fit the available widht", which is the opposite of what it means :)
Though I'm not sure what name to suggest that communicates "as narrow as possible" or "only as wide as is needed". `Narrow`?

Reply:
Maybe I'm still misunderstanding this… if buttons are used, the label will be truncated - but not otherwise? If you put a wide label in a narrow ui, will `Fit` only truncate the text if you also add buttons? That's a weird side-effect O.o

Reply:
Will `Available` ever truncate the text, or will it expand the parent ui?

Reply:
`Available` uses `ui.available_width()`, which means that button will be right aligned to the container's right border. That works for fixed-sized container like the left panel. It does *not* work for the stream, as that stream tree area is dynamically sized to its content.

This is why I introduced `Fit`. In this case, the item only allocates what it needs for the collapsing triangle (if any), the icon (if any), and the actual label. In the code, it was difficult to account for the the buttons' width, and I don't need it for the streams as that UI doesn't use buttons. So I just left that limitation.

Text truncation is somewhat orthogonal to that. Buttons will be right aligned to whatever width as been allocated. In the case of `Fit`, since the buttons are not accounted for in the allocation, they will _always_ truncate the text if used. But again, we don't need `Fit` + buttons so far. For `Available` text will be truncated if needed, either due to the fact that `ui.available_width()` is too narrow, and/or because what's left isn't sufficient when buttons shows up.